src/services/jsTyping.ts

```diff
@@ -199,17 +202,20 @@ namespace ts.JsTyping {
         }
 
         /**
-         * Infer typing names from node_module folder
-         * @param nodeModulesPath is the path to the "node_modules" folder
+         * Infer typing names from packages folder (ex: node_module, bower_components)
+         * @param packagesFolderPath is the path to the packages folder
+
          */
-        function getTypingNamesFromNodeModuleFolder(nodeModulesPath: string) {
+        function getTypingNamesFromPackagesFolder(packagesFolderPath: string) {
+            filesToWatch.push(packagesFolderPath);
+
             // Todo: add support for ModuleResolutionHost too
-            if (!host.directoryExists(nodeModulesPath)) {
+            if (!host.directoryExists(packagesFolderPath)) {
                 return;
             }
 
             const typingNames: string[] = [];
-            const fileNames = host.readDirectory(nodeModulesPath, [".json"], /*excludes*/ undefined, /*includes*/ undefined, /*depth*/ 2);
+            const fileNames = host.readDirectory(packagesFolderPath, [".json"], /*excludes*/ undefined, /*includes*/ undefined, /*depth*/ 2);
             for (const fileName of fileNames) {
                 const normalizedFileName = normalizePath(fileName);
                 if (getBaseFileName(normalizedFileName) !== "package.json") {

```

Comment:
We already retrieve the bower.json typings using getTypingsFromJson(bowerJsonPath, filesToWatch).  This addresses the case when there is no bower.json present and just a bower_components folder that contains the packages.  From my tests the bower_components folder is similar to the node_modules folder.  It contains a folder directory for each package with a package.json (bower_components/jQuery/package.json).  I've been testing using the ASP.NET UI.  Let me try bower cmd line and see if this is consistent.